examples/track.rs

```diff
@@ -12,24 +12,53 @@ impl Component for TrackedComponent {
 
 #[derive(Default)]
 struct SysA {
-    modified_id: Option<ReaderId<ModifiedFlag>>,
+    reader_id: Option<ReaderId<ComponentEvent>>,
+    events: Vec<ComponentEvent>,
+    inserted: BitSet,
     modified: BitSet,
+    removed: BitSet,
 }
 
 impl<'a> System<'a> for SysA {
     type SystemData = (Entities<'a>, ReadStorage<'a, TrackedComponent>);
 
     fn setup(&mut self, res: &mut Resources) {
         Self::SystemData::setup(res);
-        self.modified_id = Some(WriteStorage::<TrackedComponent>::fetch(&res).track_modified());
+        self.reader_id = Some(WriteStorage::<TrackedComponent>::fetch(&res).track());
     }
 
     fn run(&mut self, (entities, tracked): Self::SystemData) {
-        tracked.populate_modified(&mut self.modified_id.as_mut().unwrap(), &mut self.modified);
+        self.events  = tracked.channel().read(self.reader_id.as_mut().expect("ReaderId not found")).map(|e| *e).collect();
+
+        self.modified.clear();
+        self.modified.extend(self.events.iter().filter_map(|event| match event {
+            ComponentEvent::Modified(index) => Some(index),
+            _ => None,
+        }));
```

Comment:
These should probably be made into helper methods for `ComponentEvent`:
```rust
fn modified(&self) -> Option<Index>
```

Reply:
Actually would it be better to iterate events only once and insert to each different `BitSet`?

Reply:
I'd rather just do the latter, the "helper" methods might be kind of confusing.